src/components/PeopleTable.tsx

```diff
@@ -1,645 +1,197 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import cn from 'classnames';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PersonLink } from './PersonLink';
+import { Person } from '../types';
+import { getSearchWith } from '../utils/searchHelper';
+
+type Props = {
+  data: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ data }) => {
+  const { personSlug } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const selectedPersonSlug = personSlug ? personSlug : null;
+
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  function handleSortField(sortField: string) {
+    if (sortBy !== sortField) {
+      const newSearchParams = getSearchWith(searchParams, {
+        sort: sortField || null,
+      });
+
+      setSearchParams(newSearchParams);
+    } else if (order !== 'desc') {
+      const newSearchParams = getSearchWith(searchParams, {
+        order: 'desc',
+      });
+
+      setSearchParams(newSearchParams);
+    } else {
+      const newSearchParams = getSearchWith(searchParams, {
+        sort: null,
+        order: null,
```

Comment:
They are almost the same, only the second argument that you pass to `getSearchWith` is different. So you can prepare object and create search params only once